app/grandchallenge/reader_studies/models.py

```diff
@@ -1075,6 +1075,9 @@ class ScoringFunction(models.TextChoices):
     interface = models.ForeignKey(
         ComponentInterface, on_delete=models.PROTECT, null=True, blank=True
     )
+    widget = models.OneToOneField(
```

Comment:
Users can delete questions, so we should ensure that any widget is deleted too. You will need to use signals, see what we do with related groups for reader studies, challenges, etc, and ensure it's tested.

Reply:
ah so the `on_delete` shouldn't be `PROTECT` because that would keep them from deleting the question if a widget is set, right?

Reply:
oh no that's for the opposite way, deleting the widget wouldn't work because it's associated with a question

Reply:
Yes, could also be done the other way around - set `question = models.OneToOne(blank=False, null=False, on_delete=models.CASCADE, related_name="widget")`. That might be better actually as it always ensures that a widget has a question, but I don't think enforces that each question has a widget.

Reply:
Hm, yes, but then it would be difficult to include the widget in the question serializer, wouldn't it?

Reply:
I don't see why, you could still do `question.widget` - that is what the `related_name` is about.

Reply:
Yes, you're right. Sorry, got confused for a minute. This reverse set-up should also work just as well on the question form, so I think this a good design.